survive-the-monsters.js

```diff
@@ -1,45 +1,94 @@
 
-const actors = {};
+const actors = {
+    enemies: new Array(ENEMIES_COUNT),
+};
 
-class GameScene extends Phaser.Scene {
-    preload () {
-        this.load.image("player", "sprites/players/mvp_player/idle_0.png");
-        this.load.image("monster_0", "sprites/monsters/mvp_monster_0/idle_0.png");
-        this.load.image("monster_1", "sprites/monsters/mvp_monster_1/idle_0.png");
-    }
-    create () {
-        actors.player = this.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, "player");
-        actors.player.setVelocity(100, 0);
-        actors.player.setBounce(1, 1);
-        actors.player.setCollideWorldBounds(true);
-
-        actors.monster_0 = this.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, "monster_0");
-        actors.monster_0.setVelocity(-150, 10);
-        actors.monster_0.setBounce(1, 1);
-        actors.monster_0.setCollideWorldBounds(true);
-        
-        actors.monster_1 = this.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, "monster_1");
-        actors.monster_1.setVelocity(-200, 0);
-        actors.monster_1.setBounce(1, 1);
-        actors.monster_1.setCollideWorldBounds(true);
+const sceneCenterX = SCENE_WIDTH / 2;
+const sceneCenterY = SCENE_HEIGHT / 2;
+const enemiesX = new Float64Array(ENEMIES_COUNT);
+const enemiesY = new Float64Array(ENEMIES_COUNT);
+const input = [0, 0];
+let playerX = 0;
+let playerY = 0;
+let inputKeys;
+
+const scenes = {
+    main: new Phaser.Scene("MainScene"),
+};
+scenes.main.preload = function() {
+    this.load.image("player", "sprites/players/mvp_player/idle_0.png");
+    this.load.image("monster_0", "sprites/monsters/mvp_monster_0/idle_0.png");
+    this.load.image("monster_1", "sprites/monsters/mvp_monster_1/idle_0.png");
+};
+scenes.main.create = function() {
+    actors.player = this.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, "player");
+    for (let enemyId = 0; enemyId < ENEMIES_COUNT; enemyId++) {
+        actors.enemies[enemyId] = this.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, `monster_${enemyId}`);
     }
-}
+    
+    actors.player.setPosition(sceneCenterX, sceneCenterY);
+
+    this.physics.world.on("worldstep", physicsLoop);
+    inputKeys = this.input.keyboard.addKeys({
+        up: "W",
+        down: "S",
+        left: "A",
+        right: "D",
+    });
+};
 
 const game = new Phaser.Game({
     type: Phaser.AUTO,
     width: SCENE_WIDTH,
     height: SCENE_HEIGHT,
-    scene: GameScene,
+    scene: scenes.main,
     physics: {
         default: "arcade",
         arcade: {
             gravity: { y: GRAVITY }
         }
     }
-});
\ No newline at end of file
+});
+
+function physicsLoop(dt) {
+    updateInput();
+    normalizeInput();
+    applyMovement(dt);
+    updatePositions();
+}
+function updateInput() {
+    input[0] = Number(inputKeys.right.isDown) - Number(inputKeys.left.isDown);
+    input[1] = Number(inputKeys.down.isDown) - Number(inputKeys.up.isDown);
+}
+function normalizeInput() {
+    const mSq = input[0]**2 + input[1]**2;
+    if (mSq == 0) {
```

Comment:
**suggestion**: Consider using `Math.hypot` for calculating the magnitude, which can improve readability.